infra/base-images/base-builder/detect_repo.py

```diff
@@ -29,25 +29,37 @@
 
 
 def main():
-  """Function to get a git repos information based on its commit."""
+  """Function to get a git repos information based on its commit.
+
+  Raises:
+    ValueError when a commit or a ref is not provided
+  """
   parser = argparse.ArgumentParser(
       description='Finds a specific git repo in an oss-fuzz projects docker file.'
   )
   parser.add_argument(
       '--src_dir',
       help='The location of the oss-fuzz projects source directory',
       required=True)
+  parser.add_argument(
+      '--repo_name', help='the name of the github repo', required=False)
   parser.add_argument(
       '--example_commit',
       help='A commit SHA refrencing the projects main repo',
-      required=True)
+      required=False)
+
   args = parser.parse_args()
+  if not args.repo_name and not args.example_commit:
+    raise ValueError(
+        'Requires either a example commit or a repo name to detect repo.')
   for single_dir in os.listdir(args.src_dir):
     full_path = os.path.join(args.src_dir, single_dir)
     if not os.path.isdir(full_path):
       continue
-    if check_for_commit(
-        os.path.join(args.src_dir, full_path), args.example_commit):
+    if args.example_commit and check_for_commit(full_path, args.example_commit):
+      print('Detected repo: %s %s' % (get_repo(full_path), single_dir.rstrip()))
```

Comment:
is rstrip needed here? If so why isn't it needed on 56?

Reply:
No it is not.